common/src/file_slice.rs

```diff
@@ -32,6 +33,61 @@ pub trait FileHandle: 'static + Send + Sync + HasLen + fmt::Debug {
     }
 }
 
+#[derive(Debug)]
+/// A File with it's length included.
+pub struct WrapFile {
+    file: File,
+    len: usize,
+}
+impl WrapFile {
+    /// Creates a new WrapFile and stores its length.
+    pub fn new(file: File) -> io::Result<Self> {
+        let len = file.metadata()?.len() as usize;
+        Ok(WrapFile { file, len })
+    }
+}
+
+#[async_trait]
+impl FileHandle for WrapFile {
+    fn read_bytes(&self, range: Range<usize>) -> io::Result<OwnedBytes> {
+        let file_len = self.len();
+
+        // Calculate the actual range to read, ensuring it stays within file boundaries
+        let start = range.start;
+        let end = range.end.min(file_len);
+
+        // Ensure the start is before the end of the range
+        if start >= end {
+            return Err(io::Error::new(io::ErrorKind::InvalidInput, "Invalid range"));
+        }
+
+        let mut buffer = vec![0; end - start];
+
+        #[cfg(unix)]
+        {
+            use std::os::unix::prelude::FileExt;
+            self.file.read_exact_at(&mut buffer, start as u64)?;
+        }
+
+        #[cfg(not(unix))]
+        {
+            let mut file = self.file.try_clone()?; // Clone the file to read from it separately
+                                                   // Seek to the start position in the file
+            file.seek(io::SeekFrom::Start(start as u64))?;
+            // Read the data into the buffer
+            file.read_exact(&mut buffer)?;
+        }
```

Comment:
Any reason why we are not using [`std::os::windows::fs::FileExt`](https://doc.rust-lang.org/stable/std/os/windows/fs/trait.FileExt.html) here?

Reply:
I wasn't aware of it. 
Windows is not a priority though, except that it should work

Reply:
> Windows is not a priority though, except that it should work

Fair enough